tun_freebsd.go

```diff
@@ -0,0 +1,67 @@
+package nebula
+
+import (
+	"fmt"
+	"net"
+	"os/exec"
+	"strconv"
+
+	"github.com/yggdrasil-network/water"
+)
+
+type Tun struct {
+	Device string
+	Cidr   *net.IPNet
+	MTU    int
+
+	*water.Interface
+}
+
+func newTun(deviceName string, cidr *net.IPNet, defaultMTU int, routes []route, unsafeRoutes []route, txQueueLen int) (ifce *Tun, err error) {
+	if len(routes) > 0 {
+		return nil, fmt.Errorf("Route MTU not supported in FreeBSD")
+	}
+	if len(unsafeRoutes) > 0 {
+		return nil, fmt.Errorf("unsafeRoutes not supported in FreeBSD")
+	}
+	// NOTE: You cannot set the deviceName under FreeBSD, so you must check tun.Device after calling .Activate()
+	return &Tun{
+		Cidr: cidr,
+		MTU:  defaultMTU,
+	}, nil
+}
+
+func (c *Tun) Activate() error {
+	var err error
+	c.Interface, err = water.New(water.Config{
+		DeviceType: water.TUN,
+	})
+	if err != nil {
+		return fmt.Errorf("Activate failed: %v", err)
+	}
+
+	c.Device = c.Interface.Name()
+
+	// TODO use syscalls instead of exec.Command
+	fmt.Println("command: ifconfig", c.Device, c.Cidr.String(), c.Cidr.IP.String())
```

Comment:
remove this fmt.Println and the one below, or change it to use the logger that the rest of Nebula uses (search the repository for `l.Debug(…)` etc). Debug level would be correct here I believe.